app/form_models/publishers/job_listing/working_patterns_form.rb

```diff
@@ -0,0 +1,5 @@
+class Publishers::JobListing::WorkingPatternsForm < Publishers::JobListing::VacancyForm
+  attr_accessor :working_patterns, :working_patterns_details
+
+  validates :working_patterns, presence: true
```

Comment:
Could validate inclusion in the set of possible working patterns, too. We tend to validate inclusion for checkbox/radio button fields, I would guess 'to be on the safe side' (a user could put whatever parameters they liked into the post url, and then in a job listing we'd be rendering user-input data that we don't expect to come from a user, which could be dangerous)

Reply:
I believe I used inclusion at first but had problems with. I'll give it another go.